pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -869,26 +888,45 @@ func (a *AmbientIndex) handleService(obj any, isDelete bool, c *Controller) sets
 	return updates
 }
 
-func compareGatewayAddress(a1, a2 *workloadapi.GatewayAddress) bool {
-	if a1 == nil && a2 == nil {
-		return true
-	}
-	if (a1 == nil && a2 != nil) || (a1 != nil && a2 == nil) {
-		return false
+func convertGatewayAddress(gtw *workloadapi.GatewayAddress) string {
+	if gtw == nil {
+		return ""
 	}
-
-	if a1.Port != a2.Port {
-		return false
+	addrValue := ""
+	switch addr := gtw.Destination.(type) {
+	case *workloadapi.GatewayAddress_Address:
+		addrValue = addr.Address.Network + "/" + string(addr.Address.Address)
+	case *workloadapi.GatewayAddress_Hostname:
+		addrValue = addr.Hostname.Hostname + "." + addr.Hostname.Namespace
```

Comment:
How about cluster ? And network for that matter - is the hostname global ? 

Maybe it's better to not mix multi-network support to this PR ? It may need separate design and discussion, the old multi-network support would be very different with hbone.